### DISCUSSION TITLE
How to detect what is module is using an implicit import?

### DISCUSSION THREAD

**Discussion by bebraw** [denoland/deno#21150] [Q&A] [answered]
I noticed that when I build [my project](https://github.com/gustwindjs/gustwind) newly, it's giving the following warning:

> Warning Implicitly using latest version (0.206.0) for https://deno.land/std/path/mod.ts

Do you know how to figure out which module is causing this? I know it's because there's a module import without a version specifier somewhere but I don't know how to find it. I also know that this can be a problem if/when a module API changes as that could potentially break the import.

**Comment by 0f-0b** [comment, marked as answer]
Run `deno info <url to the entry point>` and search for `https://deno.land/std@0.206.0/path/mod.ts`.

```text
Warning Implicitly using latest version (0.206.0) for https://deno.land/std/path/mod.ts
…
│ │ ├─┬ https://deno.land/x/gustwind@v0.54.3/utilities/fs.ts (900B)
│ │ │ ├─┬ https://deno.land/x/recursive_readdir@v2.0.0/mod.ts (513B)
│ │ │ │ └─┬ https://deno.land/std@0.206.0/path/mod.ts (2.26KB)
…
```

It reveals that the offending module is `recursive_readdir`. You can replace it with [`std/fs/walk.ts`](https://deno.land/std@0.206.0/fs/walk.ts).

**Reply by bebraw** [reply]
Awesome, thanks a lot.
